### PR TITLE
feat: multi-tenant scaler and interceptor in the HTTP add-on

### PR DESCRIPTION
**Pull request by arschles**
These are the chart changes to go along with https://github.com/kedacore/http-add-on/pull/206

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/master/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Expose ports on the interceptor and build `svc-interceptor-admin.yaml` and `svc-interceptor-proxy.yaml`
- [x] Expose ports on the operator and build `svc-operator-admin.yaml`
- [x] Test chart install with new images built from https://github.com/kedacore/http-add-on/pull/206
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*
- [x] README is updated with new configuration values *(if applicable)*

Fixes #

### CODE REVIEW

File: http-add-on/templates/deployment-interceptor.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        httpscaledobjects.http.keda.sh/version: {{ .Values.images.tag | default .Chart.AppVersion }}
+        keda.sh/addon: {{ .Chart.Name }}
+        app: {{ .Chart.Name }}
+        name: {{ .Chart.Name }}-interceptor
+        app.kubernetes.io/name: {{ .Chart.Name }}-interceptor
+        app.kubernetes.io/version: {{ .Values.images.tag | default .Chart.AppVersion }}
+        app.kubernetes.io/component: interceptor
+        app.kubernetes.io/part-of: {{ .Chart.Name }}
+        app.kubernetes.io/managed-by: {{ .Release.Service }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (tomkerkhove) [inline http-add-on/templates/deployment-interceptor.yaml:39]: Should we move this to `_helpers.tpl`?

File: http-add-on/templates/deployment-interceptor.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        image: "{{ .Values.images.interceptor }}:{{ .Values.images.tag | default .Chart.AppVersion }}"
+        imagePullPolicy: '{{ .Values.interceptor.pullPolicy | default "Always" }}'
+        name: "{{ .Chart.Name }}-interceptor"
+        env:
+        # - name: KEDA_HTTP_OPERATOR_SERVICE_NAME
+        #   value: "{{ .Chart.Name }}-{{ .Values.operator.adminService }}"
+        # - name: KEDA_HTTP_OPERATOR_SERVICE_PORT
+        #   value: "{{ .Values.operator.adminPort }}"
+        # - name: KEDA_HTTP_OPERATOR_ROUTING_TABLE_PATH
+        #   value: "{{ .Values.operator.routingTableURLPath }}"
```
Comment (tomkerkhove) [inline http-add-on/templates/deployment-interceptor.yaml:53]: Do we need them? If not, let's remove them

File: http-add-on/templates/routing-table-config-map.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: keda-http-routing-table
+  labels:
+    "control-plane": "operator"
+    "keda.sh/addon": "http-add-on"
+    "app":           "http-add-on"
+    "name":          "http-add-on-routing-table"
+data:
```
Comment (tomkerkhove) [inline http-add-on/templates/routing-table-config-map.yaml:10]: On purpose as a place holder or?

File: http-add-on/values.yaml [hunk 4]
```diff
@@ -13,6 +13,37 @@ operator:
...
+  replicasWaitTimeout: 1500ms
+  minReplicas: 1
+  maxReplicas: 50
+  pollingInterval: 1
+  deploymentCachePollingIntervalMS: 1000
+  forceHTTP2: false
+  maxIdleConns: 100
+  idleConnTimeout: 90s
+  tlsHandshakeTimeout: 10s
+  expectContinueTimeout: 1s
```
Comment (tomkerkhove) [inline http-add-on/values.yaml:46]: What about this?
```suggestion
interceptor:
  pullPolicy: Always
  admin:
    service: interceptor-admin
    port: 9090
  proxy
    service: interceptor-proxy
    port: 8080
  routingTableUpdateDurationMS: 500
  tcpConnectionTimeout: 500ms
  keepAlive: 1s
  responseHeaderTimeout: 500ms
  replicasWaitTimeout: 1500ms
  replicas:
    min: 1
    max: 50
  pollingInterval: 1
  deploymentCachePollingIntervalMS: 1000
  forceHTTP2: false
  maxIdleConns: 100
  idleConnectionTimeout: 90s
  tlsHandshakeTimeout: 10s
  expectContinueTimeout: 1s
```

File: http-add-on/templates/deployment-interceptor.yaml (hunk 2 above)
Comment (arschles) [reply http-add-on/templates/deployment-interceptor.yaml]: removed in 0ac99c73b5d595d604f30d1fc665557f3e9f31b5

File: http-add-on/templates/routing-table-config-map.yaml (hunk 3 above)
Comment (arschles) [reply http-add-on/templates/routing-table-config-map.yaml]: removed in dbd741343de44b40b1b7cc1957aaa0d239a6dc0a

File: http-add-on/values.yaml (hunk 4 above)
Comment (arschles) [reply http-add-on/values.yaml]: done in https://github.com/kedacore/charts/pull/169/commits/9a86221e61f5f1ba36f8b1c5b47394288dc5134c

**Comment by arschles** [comment]
@tomkerkhove this is ready for merge! The HTTP Addon is ready to release `v0.2.0`, so when this is merge I'll hit the button over there

**Comment by tomkerkhove** [comment]
Let's do this!

**Review by tomkerkhove — APPROVED** [review]
LGTM, feel free to merge when it has been shipped

**Comment by arschles** [comment]
@tomkerkhove I don't seem to have permission to merge. Looked through my email to see if there are any invites for GitHub but don't see anything. My GitHub notifications are a bit of a nightmare, so I may have missed something.

**Comment by tomkerkhove** [comment]
PTAL

**Comment by arschles** [comment]
@tomkerkhove got the permissions now, thank you!

I'll merge this on Monday (would like to avoid a Friday release) and do the release shortly thereafter.

**Pull request merged**